packages/base/lua/OS.lua

```diff
@@ -77,6 +77,65 @@ end
 -- system & A string with the operating system.
 -- @usage print(sessionInfo().mode)
 function sessionInfo()
-	return info_ -- this is a global variable created when TerraME is initialized
+	local info_ = _Gtme.info_ -- this is a global variable created when TerraME is initialized
+	local args = {
+		mode = {"default", "debug", "normal", "quiet", "strict"},
+		dbVersion = false,
+		separator = false,
+		silent = "boolean",
+		color = "boolean",
+		fullTraceback = "boolean",
+		autoclose = "boolean",
+		system = false,
+		round = "number",
+		version = "string",
+		currentFile = "string",
+		path = function(midx, mvalue)
+			if not Directory(mvalue):exists() then
+				customError("The argument '"..midx.."' cannot be change by '"..mvalue.."'. Directory does not exist.")
+			end
+		end
+	}
+
+	local sessionInfo_ = {}
+	local metaTableSessionInfo_ = {
+		__index = function(_, idx)
+			return info_[idx]
+		end,
+		__newindex = function(_, idx, value)
+			local ok = false
+			forEachElement(args, function(arg, check)
+				if idx ~= arg then return end
+
+				if not check then
+					customError("The argument '"..idx.."' is an important information about the current execution and cannot be change.")
```

Comment:
"Argument ... is an information about the current execution and cannot be changeD."